internal/process/process.go

```diff
@@ -27,18 +27,24 @@ func ActivationPID(cfg Configurable) int32 {
 
 	procInfoErrMsgFmt := "Could not detect process information: %v"
 
+	seen := map[int32]bool{}
 	for pid != 0 && pid != ppid {
 		logging.Debug("Current PID: %d, Parent PID: %d", pid, ppid)
 		pidFileName := ActivationPIDFileName(cfg.ConfigPath(), int(pid))
 		logging.Debug("Looking for activation pid file: %s", pidFileName)
 		if fileutils.FileExists(pidFileName) {
+			logging.Debug("Found activation pid file: %s", pidFileName)
 			return pid
 		}
 		logging.Debug("Activation pid file not found")
 
 		if ppid == 0 {
+			logging.Debug("Parent PID is 0")
 			return -1
 		}
+		if seen[ppid] {
+			logging.Debug("Parent process PID has already been seen")
```

Comment:
Don't we want a `continue` here?

Reply:
Whoops missed that. I believe we want to return here. We've already seen this parent ID which means it's been recycled so we're in the loop. Also continuing here would not change the `pid` and `ppid` values so the loop would get stuck again.